src/App.js

```diff
@@ -14,11 +15,99 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    goods: goodsFromServer,
+    isVisible: true,
+  }
+
+  startGoods = () => {
+    this.setState(prevState => ({
+      isVisible: !prevState.isVisible,
+    }));
+  }
+
+  reverseGoods = () => {
+    this.setState(prevState => ({
+      goods: [...prevState.goods].reverse(),
+    }));
+  };
+
+  sortAlphabeticallyGoods = () => {
+    this.setState(prevState => ({
+      goods: [...prevState.goods].sort(
+        (prev, next) => prev.localeCompare(next),
+      ),
+    }));
+  };
+
+  resetGoods = () => {
+    this.setState(({
+      goods: goodsFromServer,
+    }));
+  };
+
+  sortByLengthGoods = () => {
+    this.setState(prevState => ({
+      goods: [...prevState.goods].sort(
+        (prev, next) => next.length - prev.length,
```

Comment:
в этом задании сортируем по возрастанию, но это не так существенно)
```suggestion
        (prev, next) => next.length - prev.length,
```
```suggestion
        (prev, next) => prev.length - next.length,
```